driver-core/src/test/java/com/datastax/driver/core/CustomTypeTest.java

```diff
@@ -22,26 +22,85 @@
 import java.util.List;
 
 import org.testng.annotations.Test;
-import static org.testng.Assert.assertEquals;
+
+import static com.google.common.collect.Lists.newArrayList;
+
+import static com.datastax.driver.core.Assertions.assertThat;
+import static com.datastax.driver.core.DataType.cint;
+import static com.datastax.driver.core.DataType.custom;
+import static com.datastax.driver.core.DataType.list;
 
 /**
  * Test we "support" custom types.
  */
 public class CustomTypeTest extends CCMBridge.PerClassSingleNodeCluster {
 
+    public static final DataType CUSTOM_DYNAMIC_COMPOSITE = custom(
+        "org.apache.cassandra.db.marshal.DynamicCompositeType("
+        + "s=>org.apache.cassandra.db.marshal.UTF8Type,"
+        + "i=>org.apache.cassandra.db.marshal.Int32Type)");
+
+    public static final DataType CUSTOM_COMPOSITE = custom(
+        "org.apache.cassandra.db.marshal.CompositeType("
+            + "org.apache.cassandra.db.marshal.UTF8Type,"
+            + "org.apache.cassandra.db.marshal.Int32Type)");
+
+    public static final DataType LIST_OF_INTS = list(cint());
+
     @Override
     protected Collection<String> getTableDefinitions() {
         return Collections.singleton(
-              "CREATE TABLE test ("
-            + "    k int,"
-            + "    c 'DynamicCompositeType(s => UTF8Type, i => Int32Type)',"
-            + "    v int,"
-            + "    PRIMARY KEY (k, c)"
-            + ") WITH COMPACT STORAGE"
+            "CREATE TABLE test ("
+                + "    k int,"
+                + "    c1 'DynamicCompositeType(s => UTF8Type, i => Int32Type)',"
+                + "    c2 'ReversedType(CompositeType(UTF8Type, Int32Type))'," // reversed translates to CLUSTERING ORDER BY DESC
+                + "    c3 'ListType(Int32Type)'," // translates to list<int>
```

Comment:
Weird error. Anyway it goes away if we change the column type to `Int32Type`, so I did that.